### PR TITLE
Implement conversions for stablehlo logistic, tan, tanh and log ops

### LINKED ISSUE #1063: [Ops] Support for tanh op (ttnn.tanh)

### PR DESCRIPTION
**Pull request by uazizTT**
Fixes #1063

### CODE REVIEW

**Review by mmanzoorTT — APPROVED** [review]
I'll suggest to add these ops in alphabetical order now and avoid refactoring later.

**Comment by uazizTT** [comment]
> I'll suggest to add these ops in alphabetical order now and avoid refactoring later.

Yes I tried with my last PR [here](https://github.com/tenstorrent/tt-mlir/pull/1122) but I found that due to large volume of PRs going in, it causes a lot of merge conflicts and also makes it difficult for reviewers to read due to code movement. Secondly, I see others merging PRs without sorting them in alphabetical order. So either the reviewers enforce it for each PR going forward to avoid repeated merged conflicts or else we can refactor once at the end.

**Review by tapspatel — APPROVED** [review]
looks great! Thank you for updating the silicon tests with your changes

**Review by jnie-TT — APPROVED** [review]

**Comment by jserbedzijaTT** [comment]
I am already implementing the log operation as described in [this issue](https://github.com/tenstorrent/tt-mlir/issues/1098). In the future, please check if there is already an issue for the operation you are implementing, as this is becoming too chaotic.

**Comment by uazizTT** [comment]
> I am already implementing the log operation as described in [this issue](https://github.com/tenstorrent/tt-mlir/issues/1098). In the future, please check if there is already an issue for the operation you are implementing, as this is becoming too chaotic.

Hey @jserbedzijaTT I think it the other way around, @mrakitaTT created the [issue](https://github.com/tenstorrent/tt-mlir/issues/1024) last week which is assigned to me. I think it is important to make sure that an issue is not created already before opening a new one @pilkicTT.

File: include/ttmlir/Dialect/TTIR/IR/TTIROps.td [hunk 1]
```diff
@@ -288,6 +288,27 @@ def TTIR_NegOp: TTIR_ElementwiseUnaryOp<"neg"> {
     }];
 }
 
+def TTIR_LogitOp: TTIR_ElementwiseUnaryOp<"logit"> {
```
Comment (sgligorijevicTT) [inline include/ttmlir/Dialect/TTIR/IR/TTIROps.td:291]: LogitOp is an unfortunate name for this, as this is the logistic(sigmoid). The name logit is usually used for a different function, as can be seen in [torch.special.logit](https://pytorch.org/docs/stable/special.html#torch.special.logit) and [ttnn.logit](https://docs.tenstorrent.com/ttnn/latest/ttnn/ttnn/logit.html#ttnn.logit) (the doc page is sparse on details but from reading the ttnn [code](https://github.com/tenstorrent/tt-metal/blob/4584bc3bfd1d4ec8ad50222e40097d2c76c83ca8/ttnn/ttnn/operations/unary.py#L507) you can see it expects to behave the same as torch) . Additionally, in the meantime somebody else landed a ttir::SigmoidOp so this Op is no longer needed. I would suggest removing it, as logit is not a very common function in models

Comment (uazizTT) [reply include/ttmlir/Dialect/TTIR/IR/TTIROps.td]: I refactored with the changes so that stablehlo logistic op will convert to ttir.sigmoid op.

**Comment by uazizTT** [comment]
Hi code owners, please review the PR and leave comments/approve it. Thank you!

File: test/ttmlir/Silicon/TTNN/simple_eltwise.mlir
Comment (sdjordjevicTT) [inline test/ttmlir/Silicon/TTNN/simple_eltwise.mlir:1]: Can you add a standalone compiler test under:
/test/ttmlir/Dialect/TTNN/eltwise/unaryin/test/ttmlir/Dialect/TTNN/eltwise)
/unary

**Review by sdjordjevicTT — APPROVED** [review]

File: test/ttmlir/Silicon/TTNN/simple_eltwise.mlir
Comment (uazizTT) [reply test/ttmlir/Silicon/TTNN/simple_eltwise.mlir:1]: Added.
